src/main/scala/io/github/kiemlicz/shelm/HelmPlugin.scala

```diff
@@ -306,34 +378,118 @@ object HelmPublishPlugin extends AutoPlugin {
       val ivy = ivySbt.value
       new ivy.Module(moduleSettings.value)
     },
-    publishConfiguration := PublishConfiguration()
-      .withResolverName(Classpaths.getPublishTo(publishTo.value).name)
-      .withArtifacts(packagedArtifacts.value.toVector)
-      .withChecksums(checksums.value.toVector)
-      .withOverwrite(isSnapshot.value)
-      .withLogging(UpdateLogging.DownloadOnly),
+
+    outstandingPublishRequests := 1, //careful
+    publishRegistries := Seq.empty,
+
+    publish := Def.sequential(
+      Def.taskIf {
+        if (publishRegistries.value.exists(_.isInstanceOf[IvyCompatibleHttpChartRepository])) {
+          streams.value.log.info("Starting Helm Charts publish to Ivy compatible repository")
+          (Helm / publish).value
```

Comment:
Please tell me if I undertand it correctly:

* OCI push happens via `helm` command
* CM (implementation of the OG Chart reporitory) push happens via hand-crafted HTTP requests
* "Ivy" (subset of the OG Chart reporitories) push happens via upstream SBT mechanisms
* Other implementations of OG chart repository are unsupported

Which means that we have 3 different approaches that don't even cover all the possible cases.

WDYT about outsourcing the job to the `helm push` command everywhere?

Reply:
Sorry, I was unaware of the state of `helm push` D:

My biggest gripe is invalid